packages/client/src/ComponentFactory.ts

```diff
@@ -0,0 +1,16 @@
+import { AuthenticationClient } from "./AuthenticationClient";
+import { AxiosFactory } from "./AxiosFactory";
+import { DirectoryClient } from "./DirectoryClient";
+import { LegalOfficer } from "./Types";
+
+export interface ComponentFactory {
+    buildAxiosFactory: () => AxiosFactory;
+    buildDirectoryClient: (directoryEndpoint: string, axiosFactory: AxiosFactory, token?: string) => DirectoryClient;
+    buildAuthenticationClient: (directoryEndpoint: string, legalOfficers: LegalOfficer[], axiosFactory: AxiosFactory) => AuthenticationClient;
+}
+
+export const DefaultComponentFactory = {
```

Comment:
Can you type this const ?
```suggestion
export const DefaultComponentFactory: ComponentFactory = {
```
(Sorry Java boy, here)

Reply:
No problem. :-)